File: Programmers/PG_멀쩡한 사각형/intact_square.py
w = 8
h = 12


def get_gcd(a, b):
    while b:
        a, b = b, a % b

    return a


gcd = get_gcd(max(w, h), min(w, h))
mw = w // gcd
mh = h // gcd

print((w * h) - ((mw + mh - 1) * gcd))

File: Programmers/Before2022/PG_멀쩡한 사각형/IntactSquare.java
public class IntactSquare {
    public static void main(String[] args) {
        System.out.println(Solution(8, 12));
    }

    private static long Solution(int w, int h) {
        long answer = 1;
        int a = Math.max(w, h);
        int b = Math.min(w, h);

        while (b != 0) {
            int tmp = a % b;
            a = b;
            b = tmp;
        }

        answer = ((long) w * (long) h) - (((long) w / a + (long) h / a - 1) * a);
        return answer;
    }
}
